### PR TITLE
nvm it only fixes asteroids calling (init) whoops

### PR DESCRIPTION
**Pull request by ZWRh3**
## About the PR 
fixes asteroid turfs calling the (init) proc

also removes unused bullet hole related vars and proc as it has been replaced by the component
## Why's this needed? 
tiny bit of performance

### CODE REVIEW

**Review by ZeWaka — COMMENTED** [review]
I fail to see how this is changing component code at all.

**Comment by ZWRh3** [comment]
> I fail to see how this is changing component code at all.

the lists that were causing the (init) proc to be called for literally every datum are a part of component code (poorly worded ik) eg. comp_lookup

**Comment by ZeWaka** [comment]
> > I fail to see how this is changing component code at all.
> 
> the lists that were causing the (init) proc to be called for literally every datum are a part of component code (poorly worded ik) eg. comp_lookup

There's no lists in the component var definitions you changed.

**Comment by ZWRh3** [comment]
> There's no lists in the component var definitions you changed.

image

File: code/obj/mining.dm [hunk 1]
```diff
@@ -1201,6 +1201,8 @@ TYPEINFO_NEW(/turf/simulated/wall/auto/asteroid)
 			edge_overlay.layer = TURF_EFFECTS_LAYER
 			edge_overlay.color = src.stone_color
 			A.UpdateOverlays(edge_overlay, "ast_edge_[get_dir(A,src)]")
+			if(!src.space_overlays)
+				src.space_overlays = list()
```
Comment (ZeWaka) [inline code/obj/mining.dm:1205]: why not do this in New(), they're going to have them anyways

File: code/obj/mining.dm [hunk 2]
```diff
@@ -1467,6 +1469,8 @@ TYPEINFO_NEW(/turf/simulated/wall/auto/asteroid)
 			edge_overlay.layer = TURF_EFFECTS_LAYER
 			edge_overlay.color = src.stone_color
 			A.UpdateOverlays(edge_overlay, "ast_edge_[get_dir(A,src)]")
+			if(!src.space_overlays)
```
Comment (ZeWaka) [inline code/obj/mining.dm:1472]: x

**Pull request merged**